File: Shift.DAL/Models/UserModels/UndergraduateData/JournalData/Report.cs
using System;

namespace Shift.DAL.Models.UserModels.UndergraduateData.JournalData
{
	using Shift.DAL.Models.University;
	using System.ComponentModel.DataAnnotations;
	using System.ComponentModel.DataAnnotations.Schema;

	public class Report
	{
		[Key]
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int Id { get; set; }

		public DateTime Date { get; set; }
		public string Result { get; set; }
		public string DepartmentHead { get; set; }

		public int? UndergraduateJournalId { get; set; }
		public virtual UndergraduateJournal UndergraduateJournal { get; set; }

		public int? ProtocolId { get; set; }
		public virtual Protocol Protocol { get; set; }
	}
}

File: Shift.Web/ClientApp/src/app/infrastracture/entities/ujournal/Report.ts
import { Protocol } from "../university/Protocol";

export class Report {
    public Id: number;
    public Date: Date;
    public Result: string;
    public DepartmentHead: string;
    public UndergraduateJournalId?: number;

    public ProtocolId: number;
    public Protocol: Protocol;
}